src/bokeh/models/plots.py

```diff
@@ -554,6 +555,17 @@ def _check_bad_extra_range_name(self) -> str | None:
     .. note:: This feature is experimental and may change in the short term.
     """)
 
+    window_axis = Nullable(Enum(WindowAxis), default=None, help="""
+    A dimension to use for windowed auto-ranging when there are data ranges
+    present on the plot. For example, if ``window_axis`` is set to the
+    value ``"x"`` then any data ranges in the y-dimension will compute their
+    auto-ranged extents using only data inside the x-range bounds of the
+    current viewport.
+
+    If set to None (the default) then auto-ranging will use all available
+    data, regardless of viewport.
+    """)
```

Comment:
Typically we would use `dimension` instead of `axis` in such contexts, e.g. in tools or `radius_dimension` in `Circle`.  Maybe there's motivation for a different nomenclature?

Reply:
I originally had `window_dimension` but I really thought that would seem confusing, so I changed it. I think "dimension" makes sense when we are more or less talking about a "direction" (e.g. `radius_dimension` tells which *direction* to measure in) but for this case I think it's more important to make a concrete association to the window *span*, which is precisely one or the other axes.

Reply:
cc @tcmetzger for any thoughts or alternate terminology ideas

Reply:
re-reading the docstring (which I wrote when the property was originally window_dimension), I would agree it seems some of the use of "dimension" there should probably be updated to match the "axis" usage